consensus/istanbul/core/core.go

```diff
@@ -43,11 +43,20 @@ func New(backend istanbul.Backend, config *istanbul.Config) Engine {
 	if err != nil {
 		log.Crit("Failed to open RoundStateDB", "err", err)
 	}
+	istanbulLogger := istanbul.NewIstLogger(
+		func() *big.Int {
+			view, err := rsdb.GetLastView()
```

Comment:
Instead of reading from the DB, can you instead read from `c.current`?

Reply:
Yeah but it involves a bit of weird flow